stock_picking_location_check/models/stock_picking.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2024 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    allow_location_discrepancy = fields.Boolean(
+        help="If enable, the location discrepancies will be allowed between picking "
+        "and stock move lines."
+    )
+
+    @api.constrains("state")
+    def _check_location_consistency(self):
+        """Check consistency of locations between picking and its stock move lines."""
+        for picking in self:
+            if picking.allow_location_discrepancy:
+                continue
+            # Determine if the picking is an immediate transfer
+            is_immediate_transfer = picking._check_immediate()
+            for move_line in picking.move_line_ids:
+                # If not an immediate transfer and qty_done is 0, skip this move line
+                if not is_immediate_transfer and move_line.qty_done <= 0:
+                    continue
```

Comment:
In what situation would we get into trouble without this?

Reply:
If we test with below situation, the validation error will be raised.
1. Create planned transfer for receipt for lot product 
2. Add move lines with the destination location that is different from picking's destination location.
3. Then, you change picking's dest location that is the same with move lines location.

Then, you will see like two move lines with different dest locations. As per odoo standard behavior, when we changes location in picking, it also changes to related moves locations and move lines location but doesn't change to `0.00` qty lines.
<img width="1438" alt="Screenshot 2024-03-14 at 23 46 35" src="https://github.com/qrtl/axls-custom/assets/45355704/9b74ed6a-1cba-4f0b-85f9-310b48e17c5c">